File: week1/118666/dayeon_ku.java
import java.util.*;

class Solution {   
    public String output(int[] scoreArr) {
        String ans = "";
        
        if (scoreArr[0] >= 0) ans += "R";
        else if (scoreArr[0] < 0) ans += "T";
        
        if (scoreArr[1] >= 0) ans += "C";
        else if (scoreArr[1] < 0) ans += "F";
        
        if (scoreArr[2] >= 0) ans += "J";
        else if (scoreArr[2] < 0) ans += "M";
        
        if (scoreArr[3] >= 0) ans += "A";
        else if (scoreArr[3] < 0) ans += "N";
        
        return ans;
    }
    
    public String solution(String[] survey, int[] choices) {
        int[] scores = {0, 0, 0, 0};
        
        for (int i=0; i < survey.length; i++) {
            char first = survey[i].charAt(0);
            int score = (4 - choices[i]);
            int idx = 0;
            
            if (survey[i].contains("R")) {
                idx = 0;
                if (first == 'T') score *= -1;
            }
            else if (survey[i].contains("C")) {
                idx = 1;
                if (first == 'F') score *= -1;
            }
            else if (survey[i].contains("J")) {
                idx = 2;
                if (first == 'M') score *= -1;
            }
            else if (survey[i].contains("A")) {
                idx = 3;
                if (first == 'N') score *= -1;
            }
            
            scores[idx] += score;        
        }
                
        return output(scores);
    }
}

File: week1/118666/dayeon_ku.py
def solution(survey, choices):
    indicator = {'R':0, 'T':0, 'C':0, 'F':0, 'J':0, 'M':0, 'A':0, 'N':0}
    
    for i, s in enumerate(survey):
        l, r = s[0], s[1]
        choice = choices[i]
                
        if choice == 1:
            indicator[l] += 3
        elif choice == 2:
            indicator[l] += 2
        elif choice == 3:
            indicator[l] += 1
        elif choice == 5:
            indicator[r] += 1
        elif choice == 6:
            indicator[r] += 2
        elif choice == 7:
            indicator[r] += 3
        else:
            continue
    
    first = 'R' if indicator['R'] >= indicator['T'] else 'T'
    second = 'C' if indicator['C'] >= indicator['F'] else 'F'
    third = 'J' if indicator['J'] >= indicator['M'] else 'M'
    fourth = 'A' if indicator['A'] >= indicator['N'] else 'N'
        
    return ''.join([first, second, third, fourth])